### PR TITLE
[FLINK-19124][datastream] Some JobClient methods are not ergonomic, require ClassLoader argument

### PR DESCRIPTION
**Pull request by SteNicholas**
## What is the purpose of the change

*Both `getAccumulators()` and `getJobExecutionResult()` result require the user to give a `ClassLoader`. `JobClient` created in a context could get the user-code `ClassLoader` by `PipelineExecutor`. `ClassLoader` parameter should be removed from those methods and require that users could give the user-code class loader when constructing a `JobClient` themselves for an already running job.*

## Brief change log

  - *`getAccumulators()` and `getJobExecutionResult()` of `JobClient` remove the user-code `ClassLoader` parameter.*
  - *Constructor of `ClusterClientJobClientAdapter`, `EmbeddedJobClient`, `PerJobMiniClusterJobClient` and `WebSubmissionJobClient` for`JobClient` implementations add `ClassLoader` parameter.*
  - *Caller of `getAccumulators()` and `getJobExecutionResult()` removes the user-code `ClassLoader` adapt to the modified API of `JobClient`.*  

## Verifying this change

  - *Test cases related to `getAccumulators()` and `getJobExecutionResult()` of `JobClient` could verify the case that removes the user-code `ClassLoader` parameter.*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (**yes** / no)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by SteNicholas** [comment]
@aljoscha I have already removed the user-code `ClassLoader` from `getAccumulators()` and `getJobExecutionResult()`, which `ClassLoader` could configure when `PipelineExecutor` executes. Could you please review this pull request?

**Comment by SteNicholas** [comment]
@aljoscha I have already removed the user-code ClassLoader from getAccumulators() and getJobExecutionResult(), which ClassLoader could configure when PipelineExecutor executes. Could you please review this pull request?

**Comment by aljoscha** [comment]
On it! 👌

File: flink-clients/src/main/java/org/apache/flink/client/deployment/application/WebSubmissionJobClient.java [hunk 1]
```diff
@@ -43,7 +43,7 @@
 
 	private final JobID jobId;
 
-	public WebSubmissionJobClient(final JobID jobId) {
+	public WebSubmissionJobClient(final JobID jobId, final ClassLoader userCodeClassloader) {
```
Comment (aljoscha) [inline flink-clients/src/main/java/org/apache/flink/client/deployment/application/WebSubmissionJobClient.java:46]: The `userCodeClassloader` is not used so the parameter doesn't need to be added.

File: flink-clients/src/main/java/org/apache/flink/client/deployment/application/executors/EmbeddedExecutorFactory.java [hunk 2]
```diff
@@ -79,9 +79,9 @@ public PipelineExecutor getExecutor(final Configuration configuration) {
 		return new EmbeddedExecutor(
 				submittedJobIds,
 				dispatcherGateway,
-				jobId -> {
+			(jobId, userCodeClassloader) -> {
```
Comment (aljoscha) [inline flink-clients/src/main/java/org/apache/flink/client/deployment/application/executors/EmbeddedExecutorFactory.java:82]: The indentation seems off here, should be on the same level as the other parameters.

**Review by aljoscha — CHANGES_REQUESTED** [review]
Excellent work! I see you also removed fields where the `ClassLoader` is now not needed anymore.

I just had two comments, once they're addressed this is ready to merge.

File: flink-clients/src/main/java/org/apache/flink/client/deployment/application/WebSubmissionJobClient.java (hunk 1 above)
Comment (SteNicholas) [reply flink-clients/src/main/java/org/apache/flink/client/deployment/application/WebSubmissionJobClient.java]: @aljoscha Yes, I would like to remove this unused `userCodeClassloader `.

File: flink-clients/src/main/java/org/apache/flink/client/deployment/application/executors/EmbeddedExecutorFactory.java (hunk 2 above)
Comment (SteNicholas) [reply flink-clients/src/main/java/org/apache/flink/client/deployment/application/executors/EmbeddedExecutorFactory.java]: @aljoscha Sorry for this indentation missing, I have already updated this indentation.

**Comment by SteNicholas** [comment]
@aljoscha I have followed up with your comments. If you are available, please help to review the changes again. Thanks.

**Comment by aljoscha** [comment]
Good work! I merged this. 👍

**Pull request closed without merging**